### ISSUE TITLE
Style/FileName: false positive when inspecting file in non-current directory

### ISSUE THREAD

**Issue by zenspider** [bbatsov/rubocop#1598]
I have this on the top of my Rakefile:

```
-*- ruby -*-
```

and rubocop warns me about using snake case for file names.

**Comment by bbatsov** [comment]
`Rakefile` is supposed to be a whitelisted name, so I'm not sure what's wrong here. Maybe some config problem? The Emacs comment has nothing do with this cop (the message is simply associated with the first line in the file). 

P.S. If you're using Emacs 24.4+ this comment is redundant.

**Comment by jonas054** [comment]
I can confirm that there's a bug here, though I have not found that the emacs mode line makes any difference. Here's what I've found.

* If I run `rubocop` and give it a directory that is not equal to, and not a subdirectory, of the current directory, then a `Rakefile` in the inspected directory is not inspected.
* If I give the absolute path to the Rakefile as an argument (still from a different directory tree), then the file is checked and `Style/FileName` reports an offense.

I think both of these are bugs. The behavior is different if I run from the inspected directory without arguments.

```
 /tmp$ find 1598 -ls
4457280    4 drwxrwxr-x   3 jonas    jonas        4096 jan 27 19:05 1598
4459479    4 drwxrwxr-x   2 jonas    jonas        4096 jan 27 19:05 1598/other
4458731    4 -rw-rw-r--   1 jonas    jonas         822 jan 27 18:58 1598/Rakefile
 /tmp$ cd 1598/other/
 /tmp/1598/other$ rubocop /tmp/1598/
Inspecting 0 files


0 files inspected, no offenses detected
 /tmp/1598/other$ rubocop /tmp/1598/Rakefile 
Inspecting 1 file
C

Offenses:

/tmp/1598/Rakefile:1:1: C: Use snake_case for source file names.
# encoding: utf-8
^

1 file inspected, 1 offense detected
 /tmp/1598/other$ cd ..
 /tmp/1598$ rubocop
Inspecting 1 file
.

1 file inspected, no offenses detected
```

**Comment by zenspider** [comment]
Ah. sorry. Feel free to change the title since the modeline has nothing to do with it. I assumed it was relevant.

**Comment by zenspider** [comment]
@jonas054 I wasn't running rubocop that way... lemme try to get a repro.
